R/fable_model.R

```diff
@@ -0,0 +1,66 @@
+#' Fable model wrapper
+#'
+#'
+#' @description Provides an interface for models from the `fable` package. Not the `ARIMA` model
+#' requires the `feast` package. If `future` is being used `fable` will require `future.apply` in
+#' order to not silently fail.
+#'
+#' @param model A `fable` model object. For  models that use a formula interface time
+#' can be accessed using `time`.
+#' @inheritParams bsts_model
+#' @return A dataframe of predictions (with columns representing the
+#'  time horizon and rows representing samples).
+#' @export
+#' @importFrom tsibble tsibble
+#' @importFrom fabletools model forecast
+#' @importFrom purrr map
+#' @importFrom dplyr bind_rows
+#' @examples
+#' ## Used on its own
+#' fable_model(y = EpiSoon::example_obs_rts[1:10, ]$rt,
+#'            model = fable::ARIMA(y ~ time),
+#'            samples = 10, horizon = 7)
+#'
+#'
+#'forecast_rt(EpiSoon::example_obs_rts[1:10, ],
+#'            model = function(...){
+#'            fable_model(model = fable::ARIMA(y ~ time), ...)},
+#'            horizon = 7, samples = 10)
+fable_model <- function(y = NULL, samples = NULL,
+                       horizon = NULL, model = NULL) {
+
+
+## Make input numeric into correct tsibble format
+timeseries <- tsibble::tsibble(y = y, time = 1:length(y), index = time)
```

Comment:
we are passing in a date column anyway, right? I had used as_tsibble(index = date, key = timeseries), to preserve the given dates. The key = timeseries argument is just omitted I guess for our purposes

Reply:
ah ok I see the date isn't returned anyway... Forget about my comment :)